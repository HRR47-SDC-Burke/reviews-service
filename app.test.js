import React from 'react';
import axios from 'axios';
import Enzyme, { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer';
import App from './client/app.jsx';
import regeneratorRuntime from "regenerator-runtime";
import Adapter from 'enzyme-adapter-react-15';
Enzyme.configure({ adapter: new Adapter() })
jest.mock('axios');

describe('Module', () => {
  it('Module renders to page', () => {
    const component = renderer.create(<App></App>);
    let reviews = component.toJSON();
    expect(reviews).toMatchSnapshot();
  });

});

describe('Get html from GET request to localhost:3000/:id', () => {
  it('Should get api data', async () => {
    const getSpy = jest.spyOn(axios, 'get');
    const reviewInstance = renderer.create(<App></App>);
    expect(getSpy).toHaveBeenCalled();
    getSpy.mockReset();
    getSpy.mockRestore();
  });
});
