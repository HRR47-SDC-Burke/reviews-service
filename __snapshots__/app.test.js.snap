// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Module Module renders to page 1`] = `
<div
  style={
    Object {
      "fontFamily": "Helvetica",
    }
  }
>
  <div
    className="review-header"
  >
    <h2
      className="sc-fznKkj jWwMA"
    >
       
      <div
        style={
          Object {
            "color": "red",
            "display": "inline-block",
          }
        }
      >
        ★
      </div>
       
      0
       (
      0
       reviews)
    </h2>
  </div>
  <div
    className="review-categories"
    style={
      Object {
        "display": "flex",
        "flexFlow": "row wrap",
      }
    }
  />
  <div
    className="sc-AxheI arqrE"
  />
  <button
    className="sc-fznyAO hsvusQ"
    onClick={[Function]}
  >
    Show all 
    0
     reviews
  </button>
</div>
`;

exports[`testing 1`] = `
<div
  style={
    Object {
      "fontFamily": "Helvetica",
    }
  }
>
  <div
    className="review-header"
  >
    <h2
      className="sc-fznKkj jWwMA"
    >
       
      <div
        style={
          Object {
            "color": "red",
            "display": "inline-block",
          }
        }
      >
        ★
      </div>
       
      0
       (
      0
       reviews)
    </h2>
  </div>
  <div
    className="review-categories"
    style={
      Object {
        "display": "flex",
        "flexFlow": "row wrap",
      }
    }
  />
  <div
    className="sc-AxheI arqrE"
  />
  <button
    className="sc-fznyAO hsvusQ"
    onClick={[Function]}
  >
    Show all 
    0
     reviews
  </button>
</div>
`;
